subprojects/packagefiles/gtest/googlemock/meson.build

```diff
@@ -13,39 +13,57 @@
 
 gmock_incdir = include_directories('include', '.', is_system: true)
 
-if host_machine.system() == 'windows' and get_option('default_library') != 'static'
-    target_type = 'static_library'
+cpp = meson.get_compiler('cpp')
+
+if cpp.get_argument_syntax() == 'msvc'
+    gmock_lib = static_library(
+        'gmock-all',
+        files('../googletest/src/gtest-all.cc', 'src/gmock-all.cc'),
+        include_directories: [gtest_incdir, gmock_incdir],
+    )
+
+    gmock_main_lib = static_library(
+        'gmock_main',
+        files('../googletest/src/gtest-all.cc', 'src/gmock-all.cc', 'src/gmock_main.cc'),
+        include_directories: [gtest_incdir, gmock_incdir],
+    )
+
+    gmock_dep = declare_dependency(
+        include_directories: [gtest_incdir, gmock_incdir],
+        link_with: gmock_lib,
+    )
+
+    gmock_main_dep = declare_dependency(
+        include_directories: [gtest_incdir, gmock_incdir],
+        link_with: gmock_main_lib,
+    )
 else
-    target_type = 'shared_library'
-endif
+    gmock_lib = library(
```

Comment:
I think this still needs to be a shared_library (?). The original issue was that when used statically, it was trivial to produce an ODR violation with googletest. See also https://github.com/mesonbuild/wrapdb/issues/1937

Reply:
testing now. https://github.com/google/googletest/blob/main/googlemock/CMakeLists.txt#L85 sets it as a static library, no?

Reply:
I guess the question is if our goal is to match the CMake configuration or not. FWIW the CMake configuration itself is community-provided, i.e. not technically the official build system for gtest. I think based off of the discussion from the linked content the maintainers suggest that shared linkage is the way to go

Not an expert in this area however, so happy to be wrong

Reply:
static build compiles and runs fine.

Reply:
Yes that won't be a problem. The issue only appears at runtime in rare cases. 

The smallest reproducer I've been able to find I posted in https://github.com/google/googletest/pull/4536#issuecomment-2148422262 - is there an easy way to test this against that?

Reply:
Ah ignore my comment above; the easiest reproducer I could find was in the issue already posted. So you tried this against that example and it ran fine? If so sure then I think it solves it, although it is entirely clear to me what would have changed between this and the original fix that prevents the ODR